11_structures/main.go

```diff
@@ -69,58 +73,518 @@ instances inside the function.
+
 	// TODO: Print the name of the person with the highest BMI in the array.
 	// Hint: create a function that takes a list of person and return the person
 	// with the highest BMI.
+	fmt.Println(getHighestBMI(randomArray))
+
+	// // TODO: Create a struct named "album" with the following fields:
+	// // - name: string
+	// // - year: int
+	// type album struct {
+	// 	name string
+	// 	year int
+	// }
+
+	// Added before main function
+
+	// // TODO: Create a struct named "artist" with the following fields:
+	// // - name: string
+	// // - members: a list of strings
+	// // - albums: a map[int]album where (int) is the year of the album
+
+	// Added before main function
 
-	// TODO: Create a struct named "album" with the following fields:
-	// - name: string
-	// - year: int
-	// TODO: Create a struct named "artist" with the following fields:
-	// - name: string
-	// - members: a list of strings
-	// - albums: a map[int]album where (int) is the year of the album
 	// TODO: Create a couple of albums and artists.
+	xulu := album{
+		name: "Xulu",
+		year: 2024,
+	}
+
+	zeppelin := album{
+		name: "Zeppelin",
+		year: 2020,
+	}
+
+	sweetSummer := album{
+		name: "Sweet Summer",
+		year: 1999,
+	}
+
+	harshWinter := album{
+		name: "Harsh Winter",
+		year: 2001,
+	}
+
+	yellow := artist{
+		name:    "Yellow",
+		members: []string{"Jay", "Otay", "Luna"},
+		albums:  map[int]album{2020: zeppelin, 2024: xulu},
+	}
+
+	blue := artist{
+		name:    "Blue",
+		members: []string{"Fox", "Dax", "Naomi", "Vitay"},
+		albums:  map[int]album{1999: sweetSummer, 2001: harshWinter},
+	}
+
+	fmt.Println(yellow, blue)
+
 	// TODO: Create a function on the artist struct that prints the list of all
 	// the albums of the artist.
+	blue.getDisco()
+	yellow.getDisco()
+
 	// TODO: Create a function that takes a list of artists and prints the first
 	// and last album ever released.
+	artistArray := []artist{
+		{
+			name:    "Yellow",
+			members: []string{"Jay", "Otay", "Luna"},
+			albums:  map[int]album{2020: zeppelin, 2024: xulu},
+		},
+		{
+			name:    "Blue",
+			members: []string{"Fox", "Dax", "Naomi", "Vitay"},
+			albums:  map[int]album{1999: sweetSummer, 2001: harshWinter},
+		},
+	}
+	// debug
+	fmt.Println("Print artistArray", artistArray)
+	fmt.Println(
+		"Print specific artistArray album map:",
+		artistArray[0].albums,
+	)
+	fmt.Println(
+		"Print specific album map:",
+		map[int]album{2020: zeppelin},
+	)
+	fmt.Println(
+		"Print specific album:",
+		zeppelin,
+	)
+	fmt.Println(
+		"Print all artistArray albums of specific artist:",
+		artistArray[0].albums,
+	)
+	// TO FIX
+	fmt.Println(
+		"Print specific artistArray album:",
+		artistArray[0].albums, map[int]album{2020: zeppelin},
+	)
+	fmt.Println(
+		"Print specific album year:",
+		zeppelin.year,
+	)
+	// fmt.Println(
+	// 	"Print specific artistArray album year:",
+	// 	artistArray[0].albums, map[int]album{2020: zeppelin.year},
+	// )
+	getFirstandLastAlbum(artistArray)
 }
```

Comment:
@nobe4 I've been trying to find a way to target the year of map of albums of the artistArray. No luck so far.
The syntax for exactly that, is somehow confusing to me right now.
Where should I look?

Reply:
Have a look at https://go.dev/tour/moretypes/16

Reply:
I read that, experimented a bit, but am not sure yet how to compare the album maps (or rather album years?) in the loop.
So far I wasn't able to define the correct syntax for my variable `firstAlbumYear`.
Do I have to use `reflect.DeepEqual` or something like that?

Reply:
Each artist has a map of `year => album`, if you loop with `range`, you will get both values for each entry in the map. You can then compare `year` with a `earliestYear` and `latestYear` and update variables accordingly. The comparison is between two `int`s

Reply:
This means that you have to do 2 loops: one that goes through each artist, and one that goes through each album for each artist.

Reply:
I believe I have a working logic now.